### PR TITLE
CUDA 11.1 Windows builder

### PR DESCRIPTION
**Pull request by zasdfgbnm**
How is windows binaries built? Do we need nvprune?

### CODE REVIEW

**Comment by seemethere** [comment]
@peterjc123 would probably know more

**Comment by peterjc123** [comment]
@zasdfgbnm No, we don't use nvprune on Windows.

**Comment by peterjc123** [comment]
@zasdfgbnm Will do

**Comment by peterjc123** [comment]
@zasdfgbnm The latest driver I can find for Windows is 452.39. Could you please tell me where could I download that?

**Comment by peterjc123** [comment]
@zasdfgbnm https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#cuda-whats-new-11-1-Upd1 So I assume 452.39 is enough.

**Comment by peterjc123** [comment]
@zasdfgbnm  https://ossci-windows.s3.amazonaws.com/452.39-data-center-tesla-desktop-win10-64bit-international.exe

**Comment by peterjc123** [comment]
Looks like the current driver is still not okay for CUDA 11.1.

File: windows/internal/driver_update.bat [hunk 1]
```diff
@@ -1,8 +1,8 @@
-set "DRIVER_DOWNLOAD_LINK=https://s3.amazonaws.com/ossci-windows/451.82-tesla-desktop-winserver-2019-2016-international.exe"
-curl --retry 3 -kL %DRIVER_DOWNLOAD_LINK% --output 451.82-tesla-desktop-winserver-2019-2016-international.exe
+set "DRIVER_DOWNLOAD_LINK=https://ossci-windows.s3.amazonaws.com/452.39-data-center-tesla-desktop-win10-64bit-international.exe"
+curl --retry 3 -kL %DRIVER_DOWNLOAD_LINK% --output 452.39-data-center-tesla-desktop-win10-64bit-international.exe
```
Comment (zasdfgbnm) [inline windows/internal/driver_update.bat:2]: @peterjc123 Looks like this version is working? https://github.com/pytorch/pytorch/pull/48469#issuecomment-734108204

Comment (peterjc123) [reply windows/internal/driver_update.bat:2]: Nope. Actually we are using the driver in the CUDA package.

Comment (zasdfgbnm) [reply windows/internal/driver_update.bat:2]: Oh, I see, it is at https://github.com/pytorch/pytorch/pull/48469/files#diff-54963e0b8298be39f2fb1705ffd5b9ef4043ad70226ee3a837f6d5e80aa53d74L339

Comment (peterjc123) [reply windows/internal/driver_update.bat:2]: Would you please change that?

File: windows/internal/cuda_install.bat [hunk 2]
```diff
@@ -107,6 +108,23 @@ if not exist "%SRC_DIR%\temp_build\cudnn-11.0-windows-x64-v8.0.4.30.zip" (
...
 
+:cuda111
+
+if not exist "%SRC_DIR%\temp_build\cuda_11.1.0_456.43_win10.exe" (
+    curl -k -L https://ossci-windows.s3.amazonaws.com/cuda_11.1.0_456.43_win10.exe --output "%SRC_DIR%\temp_build\cuda_11.1.0_456.43_win10.exe"
+    if errorlevel 1 exit /b 1
+    set "CUDA_SETUP_FILE=%SRC_DIR%\temp_build\cuda_11.1.0_456.43_win10.exe"
+    set "ARGS=nvcc_11.1 cuobjdump_11.1 nvprune_11.1 nvprof_11.1 cupti_11.1 cublas_11.1 cublas_dev_11.1 cudart_11.1 cufft_11.1 cufft_dev_11.1 curand_11.1 curand_dev_11.1 cusolver_11.1 cusolver_dev_11.1 cusparse_11.1 cusparse_dev_11.1 npp_11.1 npp_dev_11.1 nvrtc_11.1 nvrtc_dev_11.1 nvml_dev_11.1"
+)
+
```
Comment (peterjc123) [inline windows/internal/cuda_install.bat:119]: Actually we only need to append the optional argument here. You may write sth like 
```cmd
if "%INSTALL_DRIVER%" == "1" (
  set "ARGS=%ARGS% Display.Driver"
)
```

File: windows/internal/driver_update.bat (hunk 1 above)
Comment (zasdfgbnm) [reply windows/internal/driver_update.bat:2]: Changed in https://github.com/pytorch/builder/pull/574/files#diff-59686e8c8da2a64d7f4ddebde6c4e5a0231215eda025e43e4f3311a4ceff27d3

File: windows/internal/cuda_install.bat (hunk 2 above)
Comment (zasdfgbnm) [reply windows/internal/cuda_install.bat:119]: I think at this line, it is already guaranteed that `%CUDA_VER%` is 111? So I just added
```
if "%JOB_EXECUTOR%" == "windows-with-nvidia-gpu" set "ARGS=%ARGS% Display.Driver"
```

File: windows/internal/smoke_test.bat [hunk 3]
```diff
@@ -3,7 +3,7 @@ set SRC_DIR=%~dp0
 pushd %SRC_DIR%\..
 
 if "%CUDA_VERSION%" == "102" call internal\driver_update.bat
-if "%CUDA_VERSION%" == "110" call internal\driver_update.bat
+@REM if "%CUDA_VERSION%" == "110" call internal\driver_update.bat
```
Comment (peterjc123) [inline windows/internal/smoke_test.bat:6]: Why do we comment out this line?

Comment (zasdfgbnm) [reply windows/internal/smoke_test.bat]: You are right, it's not needed. I was thinking about something wrong.

Comment (zasdfgbnm) [reply windows/internal/smoke_test.bat]: I reverted the change

**Review by peterjc123 — APPROVED** [review]

**Comment by peterjc123** [comment]
@soumith @malfet @seemethere

**Review by malfet — APPROVED** [review]

**Comment by peterjc123** [comment]
@malfet Any blocking for this PR?

**Comment by malfet** [comment]
@peterjc123 , merged!

**Pull request merged**